src/libraries/System.Data.OleDb/src/System/Data/Common/AdapterUtil.cs

```diff
@@ -1261,10 +1261,10 @@ internal static IntPtr IntPtrOffset(IntPtr pbase, int offset)
                 return (IntPtr)checked(pbase.ToInt32() + offset);
             }
             Debug.Assert(8 == ADP.PtrSize, "8 != IntPtr.Size");
-            return (IntPtr)checked(pbase.ToInt64() + offset);
+            return (nint)(pbase.ToInt64() + offset);
```

Comment:
This is simpler and matches the behavior:
```suggestion
            return pbase + offset;
```

Reply:
Same comment as https://github.com/dotnet/runtime/pull/75557/files#r970051440